python/lsst/pipe/tasks/calibrate.py

```diff
@@ -231,7 +237,8 @@ def __init__(self, icSourceSchema=None, **kwargs):
             See also config field `icSourceFieldsToCopy`.
         @param[in,out] kwargs  other keyword arguments for lsst.pipe.base.CmdLineTask
         """
-        pipeBase.Task.__init__(self, **kwargs)
+        butler = kwargs.pop('butler')
```

Comment:
Please use butler in the list of arguments, instead. It's makes the argument list much easier to understand. I do realize you are trying to make it required. You can list it before icSourceSchema with no default, or list it after icSourceSchema with default = None and then raise an exception if butler is None. I would probably do the former, for clarity (despite the fact that it changes the argument order, so you'll have to check all callers and make sure they are providing the schema by name).

 Also please list that argument in the constructor doc string and add a @params icSourceSchema and butler.